ddtrace/contrib/django/patch.py

```diff
@@ -106,17 +106,19 @@ def _set_request_tags(django, span, request):
 
     user = getattr(request, "user", None)
     if user is not None:
-        if hasattr(user, "is_authenticated"):
-            span.set_tag("django.user.is_authenticated", user_is_authenticated(user))
-
-        uid = getattr(user, "pk", None)
-        if uid:
-            span.set_tag("django.user.id", uid)
-
-        if config.django.include_user_name:
-            username = getattr(user, "username", None)
-            if username:
-                span.set_tag("django.user.name", username)
+        # Note: getattr calls to user / user_is_authenticated may result in ImproperlyConfigured exceptions from Django
+        try:
+            if hasattr(user, "is_authenticated"):
+                span.set_tag("django.user.is_authenticated", user_is_authenticated(user))
+            uid = getattr(user, "pk", None)
+            if uid:
+                span.set_tag("django.user.id", uid)
+            if config.django.include_user_name:
+                username = getattr(user, "username", None)
+                if username:
+                    span.set_tag("django.user.name", username)
+        except django.core.exceptions.ImproperlyConfigured:
```

Comment:
```suggestion
        except Exception:
```

I think it's plausible that other types of errors might be raised while accessing the user/authentication information so I'd be in favour of making this generic.